crates/edr_evm/src/state/trie/account.rs

```diff
@@ -1,317 +1,93 @@
-use std::{collections::BTreeMap, fmt::Debug, sync::Arc};
+use std::{collections::BTreeMap, fmt::Debug};
 
-use alloy_rlp::Decodable;
-use cita_trie::{MemoryDB, PatriciaTrie, Trie as CitaTrie};
 use edr_eth::{account::BasicAccount, Address, B256, U256};
 use hasher::{Hasher, HasherKeccak};
 use revm::primitives::{Account, AccountInfo, HashMap};
+use rpds::HashTrieMapSync;
 
-/// A change to the account, where `None` implies deletion.
-pub type AccountChange<'a> = (&'a Address, Option<(BasicAccount, &'a HashMap<U256, U256>)>);
+use crate::state::trie::{
+    state_trie::{StateTrie, StateTrieMutation},
+    storage_trie::StorageTrie,
+};
 
-type AccountStorageTries = HashMap<Address, (Arc<MemoryDB>, B256)>;
-
-type Trie = PatriciaTrie<MemoryDB, HasherKeccak>;
+type StorageTries = HashTrieMapSync<Address, StorageTrie>;
 
 /// A trie for maintaining the state of accounts and their storage.
-#[derive(Debug)]
+#[derive(Clone, Debug, Default)]
 pub struct AccountTrie {
-    state_root: B256,
-    state_trie_db: Arc<MemoryDB>,
-    storage_trie_dbs: AccountStorageTries,
+    state_trie: StateTrie,
+    storage_tries: StorageTries,
 }
 
-impl AccountTrie {
+impl<'a> AccountTrie {
```

Comment:
Request for change:

The lifetime doesn't have to be on the `impl` block, as it's not used on the type. You can have `<'a>` on the individual functions that need them

Reply:
Thanks, it was a left-over from a previous iteration. Fixed now